packages/flutter/lib/src/material/input.dart

```diff
@@ -109,6 +109,7 @@ class _InputState extends State<Input> {
                                    _editableString.selection.end);
     } else if (!focused && _keyboardHandle.attached) {
       _keyboardHandle.release();
+      _editableString.composing = TextRange.empty;
```

Comment:
Yeah, I wanted to write this line of code correctly, which required refactoring how to get an empty text range.